lib/src/models/screenset_event.dart

```diff
@@ -13,6 +13,19 @@ class ScreensetEvent {
     return ScreensetEvent._(resolvedType, data);
   }
 
+  /// Construct a new [ScreensetEvent] from the given, loosely-typed [map].
+  ///
+  /// The map is expected to have an `event` key, denoting the name of the event.
+  /// The map can have a `data` key, which is a [Map] that contains any data for the event.
+  factory ScreensetEvent.fromMap(Map<Object?, Object?> map) {
```

Comment:
Adding this constructor makes the `yield` implementation a bit cleaner, since we can just rely on a tear-off and let `dart format` format the call-chain for us